src/python/pants/backend/jvm/tasks/nailgun_task.py

```diff
@@ -19,12 +19,19 @@
 
 class NailgunTaskBase(JvmToolTaskMixin, TaskBase):
   ID_PREFIX = 'ng'
+  # Possible execution strategies:
+  NAILGUN = 'nailgun'
+  SUBPROCESS = 'subprocess'
 
   @classmethod
   def register_options(cls, register):
     super(NailgunTaskBase, cls).register_options(register)
     register('--use-nailgun', type=bool, default=True,
```

Comment:
We should deprecate this option, make sure that at most one of them is set, and set the other based on its value